beacon-chain/blockchain/service.go

```diff
@@ -2,25 +2,32 @@ package blockchain
 
 import (
 	"context"
-
 	"github.com/prysmaticlabs/prysm/beacon-chain/database"
+	"github.com/prysmaticlabs/prysm/beacon-chain/powchain"
+	"github.com/prysmaticlabs/prysm/beacon-chain/types"
 	log "github.com/sirupsen/logrus"
 )
 
 // ChainService represents a service that handles the internal
 // logic of managing the full PoS beacon chain.
 type ChainService struct {
-	ctx      context.Context
-	cancel   context.CancelFunc
-	beaconDB *database.BeaconDB
-	chain    *BeaconChain
+	ctx         context.Context
+	cancel      context.CancelFunc
+	beaconDB    *database.BeaconDB
+	chain       *BeaconChain
+	web3Service *powchain.Web3Service
+	beaconBlock chan *types.Block
```

Comment:
What is this? Is this when a newly proposed block is received? Can we rename to latestBeaconBlock?